scenes/components/attack_manager.gd

```diff
@@ -0,0 +1,12 @@
+extends Node2D
+
+class_name AttackManager
```

Comment:
This feels more like an attack instance. I'm not sure what this script is trying to accomplish.